src/services/refactors/convertArrowFunctionOrFunctionExpression.ts

```diff
@@ -178,7 +178,10 @@ namespace ts.refactor.convertArrowFunctionOrFunctionExpression {
 
         const { variableDeclaration, variableDeclarationList, statement, name } = variableInfo;
         suppressLeadingTrivia(statement);
-        const newNode = factory.createFunctionDeclaration(func.decorators, statement.modifiers, func.asteriskToken, name, func.typeParameters, func.parameters, func.type, body);
+
+        const modifiersFlags = getCombinedModifierFlags(variableDeclaration) | getEffectiveModifierFlags(func);

```

Comment:
I don't have an intuition on this one - @rbuckton is this necessary?

Reply:
@rbuckton can you weigh in on this? I think this is ready to go otherwise.

Reply:
So I took a brief look at it - what sort of modifiers do you need to propagate from the variable declaration?

Reply:
`export`, basically.

Reply:
The only other one we would possibly pick up is `declare`, which is valid on a function though the initializer is an error. The result would be a `declare async function` with a body which is *also* an error.

Reply:
For the sake of clarity, and to future proof against other possibly invalid flags, perhaps this should be changed to:

```suggestion
        const modifiersFlags = (getCombinedModifierFlags(variableDeclaration) & ModifierFlags.Export) | getEffectiveModifierFlags(func);
```

It would remove `declare` in the following case, which arguably is incorrect anyways:
```ts
// before
export declare const x = async () => {}; // error on const initializer in an ambient context
// after
export async function x() {}
```